pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -417,20 +394,38 @@ func (r *ReconcilePolicyRecommendation) Reconcile(ctx context.Context, request r
 
 		certificateManager.AddToStatusManager(r.status, helper.InstallNamespace())
 
-		policyRecommendationCfg.TrustedBundle = trustedBundle
-		policyRecommendationCfg.PolicyRecommendationCertSecret = policyRecommendationKeyPair
-
-		components = append(components,
-			rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
-				Namespace:       helper.InstallNamespace(),
-				TruthNamespace:  helper.TruthNamespace(),
-				ServiceAccounts: []string{render.PolicyRecommendationName},
-				KeyPairOptions: []rcertificatemanagement.KeyPairOption{
-					rcertificatemanagement.NewKeyPairOption(policyRecommendationKeyPair, true, true),
-				},
-				TrustedBundle: trustedBundle,
-			}),
-		)
+		if !r.multiTenant {
+			trustedBundleRW := certificateManager.CreateTrustedBundle(managerInternalTLSSecret, linseedCertificate)
+			components = append(components,
+				rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
+					Namespace:       helper.InstallNamespace(),
+					TruthNamespace:  helper.TruthNamespace(),
+					ServiceAccounts: []string{render.PolicyRecommendationName},
+					KeyPairOptions: []rcertificatemanagement.KeyPairOption{
+						rcertificatemanagement.NewKeyPairOption(policyRecommendationKeyPair, true, true),
+					},
+					TrustedBundle: trustedBundleRW,
+				}),
+			)
+			trustedBundle = trustedBundleRW.(certificatemanagement.TrustedBundleRO)
+		} else {
+			trustedBundle, err = certificateManager.LoadTrustedBundle(ctx, r.client, helper.InstallNamespace())
+			if err != nil {
+				r.status.SetDegraded(operatorv1.ResourceReadError, "Error getting trusted bundle", err, logc)
+				return reconcile.Result{}, err
+			}
+			components = append(components,
+				rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
+					Namespace:       helper.InstallNamespace(),
+					TruthNamespace:  helper.TruthNamespace(),
+					ServiceAccounts: []string{render.PolicyRecommendationName},
+					KeyPairOptions: []rcertificatemanagement.KeyPairOption{
+						rcertificatemanagement.NewKeyPairOption(policyRecommendationKeyPair, true, true),
+					},
+					TrustedBundle: nil,
```

Comment:
Is nil the right value here? I'm not sure what the downstream result of this value being nil would be when setting these environment variables: https://github.com/tigera/operator/blob/master/pkg/render/policyrecommendation.go#L303

I think the difference you're intending is that in multi-tenant mode we simply load the existing trusted bundle created by the tenant controller, while in single or zero-tenant mode we need to create a trusted bundle, but in both cases I think we still need to populate the `TrustedBundle` field no?

Reply:
For multi-tenant, we should leave this nil because we want to create the secrets that contain the x509 certificates, but not the actual config map `tigera-ca-bundle`. For zero-tenant/single-tenant we want both the secrets and the config map created, because we install in tigera-policy-recommendation namespace.

Reply:
Oh so when those environment variables are blank it will trigger creation of the x509 certificates?

Reply:
Actually, the secrets that hold the certificates.